src/main/java/thecodewarrior/hooked/common/HookTickHandler.kt

```diff
@@ -155,10 +155,13 @@ object HookTickHandler {
                 hook.status = EnumHookStatus.TORETRACT
             }
 
-            if (hook.block != null) {
-                if (hook.status == EnumHookStatus.PLANTED && entity.world.isAirBlock(hook.block)) {
-                    hook.status = EnumHookStatus.TORETRACT
-                }
+            /*
+             * todo someone with more experience with Kotlin null checking and coercing needs to look at this
+             * and determine how to resolve this. Nullable instances shouldn't be passable to java methods
+             * because you can't guarantee the null safety, but somehow, it's valid code...
+             * */
+            if (hook.block != null && hook.status == EnumHookStatus.PLANTED && entity.world.isAirBlock(hook.block)) {
```

Comment:
Is this about `isAirBlock`? If so it's probably because Forge and MCP have a global `@ParametersAreNonnullByDefault`, so it's safe to just suppress it. As to why it's still valid, the nullability of Java types aren't enforced, so it can't be trusted. Thus there's a special kind of "maybe null, maybe not" type for along this boundary called a [platform type.](https://kotlinlang.org/docs/reference/java-interop.html#null-safety-and-platform-types) It's an accepted risk when using Kotlin and Java together that NPEs can still occur along the boundary between them